layers/exporters/matrix_gen.py

```diff
@@ -0,0 +1,245 @@
+from taxii2client import Server, Collection
+
+class MatrixEntry:
+    def __init__(self, id=None, name=None):
+        if id is not None:
+            self.id = id
+        if name is not None:
+            self.name = name
+
+    @property
+    def id(self):
+        if self.__id is not None:
+            return self.__id
+
+    @id.setter
+    def id(self, id):
+        self.__id = id
+
+    @property
+    def name(self):
+        if self.__name is not None:
+            return self.__name
+
+    @name.setter
+    def name(self, name):
+        self.__name = name
+
+class MatrixGen:
+    convert = {
+             'initial-access': 'Initial Access',
+             'execution': 'Execution',
+             'persistence': 'Persistence',
+             'privilege-escalation': 'Privilege Escalation',
+             'defense-evasion': 'Defense Evasion',
+             'credential-access': 'Credential Access',
+             'discovery': 'Discovery',
+             'lateral-movement': 'Lateral Movement',
+             'collection': 'Collection',
+             'command-and-control': 'Command and Control',
+             'exfiltration': 'Exfiltration',
+             'impact': 'Impact'
+        }
+    t_lookup = {'Initial Access': 'TA0001',
+                'Execution': 'TA0002',
+                'Persistence': 'TA0003',
+                'Privilege Escalation': 'TA0004',
+                'Defense Evasion': 'TA0005',
+                'Credential Access': 'TA0006',
+                'Discovery': 'TA0007',
+                'Lateral Movement': 'TA0008',
+                'Collection': 'TA0009',
+                'Command and Control': 'TA0011',
+                'Exfiltration': 'TA0010',
+                'Impact': 'TA0040'}
+    def __init__(self, fresh=True):
+        """
+            Initialization - Creates a matrix generator object
+
+            :param fresh: Bool for whether or not to read from the cti-taxii server (server currently has no
+                            subtechniques)
+        """
+        if fresh:
+            self.server = Server('https://cti-taxii.mitre.org/taxii')
+            self.api_root = self.server.api_roots[0]
+            self.collections = dict()
+
+            for collection in self.api_root.collections:
+                if collection.title != "PRE-ATT&CK":
+                    tc = Collection('https://cti-taxii.mitre.org/stix/collections/' + collection.id)
+                    self.collections[collection.title.split(' ')[0].lower()] = TAXIICollectionSource(tc)
+        else:
+            self.collections = dict()
+            stix_e = requests.get("https://raw.githubusercontent.com/mitre/cti/subtechniques/enterprise-attack"
+                                  "/enterprise-attack.json", verify=True).json()
+            stix_m = requests.get("https://raw.githubusercontent.com/mitre/cti/subtechniques/mobile-attack/mobile"
+                                  "-attack.json", verify=True).json()
```

Comment:
I don't think `fresh` is the right word for the parameter either. If the user wants to use local data, they would pass in `local_stix={enterprise, mobile}`. If local_stix is not specified then it would default to using the TAXII server.

Reply:
I've set it up to support all 3, and we can drop requests when the time arrises.